src/main/java/ch/njol/skript/expressions/ExprExperience.java

```diff
@@ -95,42 +98,50 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
 	
 	@Override
 	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
-		double d;
-		if (e instanceof ExperienceSpawnEvent)
-			d = ((ExperienceSpawnEvent) e).getSpawnedXP();
-		else if (e instanceof BlockBreakEvent)
-			d = ((BlockBreakEvent) e).getExpToDrop();
-		else
+		double eventExp;
+		if (e instanceof ExperienceSpawnEvent) {
+			eventExp = ((ExperienceSpawnEvent) e).getSpawnedXP();
+		} else if (e instanceof BlockBreakEvent) {
+			eventExp = ((BlockBreakEvent) e).getExpToDrop();
+		} else if (e instanceof PlayerExpChangeEvent) {
+			eventExp = ((PlayerExpChangeEvent) e).getAmount();
+		} else {
 			return;
-		
-		if (delta != null)
-			for (final Object o : delta) {
-				final double v = o instanceof Experience ? ((Experience) o).getXP() : ((Number) o).doubleValue();
+		}
+		if (delta == null) {
```

Comment:
This may cause unexpected behavior is `delta` is null but the change mode is something like `SET`
I would add something like this as to the beginning of the method (before anything else):
```java
if (delta == null && mode != ChangeMode.DELETE)
	return;
```

Reply:
`delta` can't be null if the mode is `SET`, it calls `change` with `DELETE` instead.

As shown here:
![image](https://user-images.githubusercontent.com/98935832/219715767-3a0f8bb1-3bed-41a6-a5d8-f12a9fc3896c.png)